### PR TITLE
Adds tooltip to semantics node

### LINKED ISSUE #86577: [a11y] avoid TYPE_WINDOW_STATE_CHANGED to announce tooltips
See https://github.com/flutter/engine/blob/eaf70ca01903d25f2cd7aad2f0f9ce754e1e0e62/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L268-L273.

As of Android 11 newer versions of TalkBack may not announce the message provided to the TYPE_WINDOW_STATE_CHANGED event. The alternative way for announcing tooltips is TBD. We may have to look at the Android source to see how the a11y for the native tooltip works or reverse-engineer the right way from TalkBack's debug logs when activating a native tooltip.

### PR DESCRIPTION
**Pull request by chunhtai**
engine part: https://github.com/flutter/engine/pull/27893

fixes https://github.com/flutter/flutter/issues/86577

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by chunhtai** [comment]
This pr needs engine change to pass the presubmit

File: packages/flutter/lib/src/semantics/semantics.dart
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart:1145]: Somewhere we should document that on iOS this is just read out as part of the label (or value?).

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by chunhtai** [comment]
google test is failing because the engine change has not rolled into google

**Comment by chunhtai** [comment]
needs to wait for the commit to roll in
https://github.com/flutter/flutter/commit/8fe859fb976cb82de21cfd2894ee874065366ba8

**Comment by chunhtai** [comment]
This is currently blocked on internal code migration

**Review by blasten — APPROVED** [review]

**Comment by Hixie** [comment]
What's the status of this PR?

**Comment by chunhtai** [comment]
I almost got to it last time and high priority thing piles up. let me close this until i have time to come back to this pr

**Comment by chunhtai** [comment]
There is no code change since the last approval, I will continue to merge this pr assuming the previous approvals still valid

**Pull request merged**
